### PR TITLE
Ensure that SunpyDeprecationWarnings are raised as errors in the test suite

### PR DESCRIPTION
**Pull request by Cadair**
### Description

It seems that at some point in the past we stopped raising SunpyDeprecationWarnings as errors in the test suite. This is *bad* :sob:

### CODE REVIEW

**Comment by Cadair** [comment]
There was one real warning in the online build, the html build fails looked like server issues.

File: examples/units_and_coordinates/SDO_to_STEREO_Coordinate_Conversion.py [hunk 1]
```diff
@@ -54,13 +54,11 @@
...
 ###############################################################################
 # We are now going to pick out a region around the south west corner:
-aia_width = 200 * u.arcsec
-aia_height = 250 * u.arcsec
 aia_bottom_left = SkyCoord(-800 * u.arcsec,
                            -300 * u.arcsec,
                            frame=maps['AIA'].coordinate_frame)
-aia_top_right = SkyCoord(aia_bottom_left.Tx + aia_width,
-                         aia_bottom_left.Ty + aia_height,
+aia_top_right = SkyCoord(-600 * u.arcsec,
```
Comment (wtbarnes) [inline examples/units_and_coordinates/SDO_to_STEREO_Coordinate_Conversion.py:60]: Maybe just get rid of `aia_top_right` and just use `aia_width` and `aia_height`?

File: sunpy/instr/tests/test_aia.py [hunk 2]
```diff
@@ -21,7 +22,8 @@ def original(request):
 
 @pytest.fixture
 def prep_map(original):
-    return aiaprep(original)
+    with pytest.warns(SunpyDeprecationWarning):
```
Comment (wtbarnes) [inline sunpy/instr/tests/test_aia.py:25]: What does this do in the fixture? Just catches the warning so pytest doesn't raise an error for this deprecation warning?

Comment (Cadair) [reply sunpy/instr/tests/test_aia.py:25]: yep

File: examples/units_and_coordinates/SDO_to_STEREO_Coordinate_Conversion.py (hunk 1 above)
Comment (Cadair) [reply examples/units_and_coordinates/SDO_to_STEREO_Coordinate_Conversion.py:60]: It was marginally easier lower down to do it this way around. :shrug:

File: sunpy/instr/tests/test_aia.py (hunk 2 above)
Comment (dstansby) [reply sunpy/instr/tests/test_aia.py:25]: And also fails the test if the code *doesn't* raise a warning.

Comment (wtbarnes) [reply sunpy/instr/tests/test_aia.py:25]: Ah ok. So works the same in a fixture as in a normal test 👍

**Pull request merged**